tutorials/foundry-start-to-end.md

```diff
@@ -0,0 +1,488 @@
+
+Now let's look at the logic in the contract. There is a single function, `run`, which is where the script logic is hosted. In this `run` function, the `vm` object is used often. This is where all of the Forge cheatcodes are stored, which determines the state of the virtual machine that the solidity is run in.  
+
+In the first line within the `run` function, `vm.envUint` is used to get a private key from the system's environment variables (we will do this soon). In the second line, `vm.startBroadcast` starts a broadcast, which indicates that the following logic should take place on-chain. So when the `MyToken` and the `Container` contracts are instantiated with the `new` keyword, they are instantiated on-chain. The final line, `vm.stopBroadcast` ends the broadcast.  
+
+Before we run this script, let's set up some of our environment variables. Create a new `.env` file:  
+
+```
+touch .env
+```
+
+And within this file, add the following:  
+
+```
+PRIVATE_KEY=YOUR_PRIVATE_KEY
+MOONSCAN_API_KEY=YOUR_MOONSCAN_API_KEY
+```
+
+!!! note
+    Foundry provides [additional options for handling your private key](https://book.getfoundry.sh/reference/forge/forge-script#wallet-options---raw){target=_blank}. It is up to you to decide whether or not you would rather use it in the console, have it stored in your device's environment, using a hardware wallet, or using a keystore.
+
+To add these environment variables, run the following command:  
+
+```
+source .env
+```
+
+Now your script and project should be ready for deployment! Use the following command to do so:  
+
+```
+forge script Container.s.sol:ContainerDeployScript --broadcast --verify -vvvv --rpc-url moonbase
+```
+
+What this command does is run the `ContainerDeployScript` contract as a script. The `--broadcast` option tells Forge to allow broadcasting of transactions, the `--verify` option tells Forge to verify to Moonscan when deploying, `-vvvv` makes the command output verbose, and `--rpc-url moonbase` sets the network to what `moonbase` was set to in `foundry.toml`.  
+
+You should see something like this as output:  
+
+![Running a Script in Foundry](/images/tutorials/foundry-start-to-end/foundry-5.png)
+
+You should be able to see that your contracts were deployed, and are verified on Moonscan! For example, this is where my [`Container.sol` contract was deployed](https://moonbase.moonscan.io/address/0xe8bf2e654d7c1c1ba8f55fed280ddd241e46ced9#code){target=_blank}.  
+
+The entire deployment script is [available on GitHub](https://raw.githubusercontent.com/PureStake/moonbeam-docs/master/.snippets/code/tutorials/foundry-start-to-end/Container.s.sol){target=_blank}.  
+
+### Deploying on Moonbeam MainNet {: #deploying-on-moonbeam-mainnet }
+
+Let's say that you're comfortable with your smart contracts and you want to deploy on the Moonbeam MainNet! The process isn't too different from what was just done, you just have to change the command's rpc-url: 
```

Comment:
I would also make a note that they'll need to update their configs from `"moonbase"` to `"moonbeam"`

Reply:
even though it's obvious 😅